woodwork/table_accessor.py

```diff
@@ -690,11 +728,11 @@ def rename(self, columns, inplace=False):
             if _is_koalas_dataframe(self._dataframe):
                 raise ValueError('Rename inplace not supported for Koalas')
             self._dataframe.rename(columns=columns, inplace=True)
-            self.init(schema=new_schema)
+            self.init_with_full_schema(schema=new_schema)
             return
         new_df = self._dataframe.rename(columns=columns)
 
-        new_df.ww.init(schema=new_schema)
+        new_df.ww.init_with_full_schema(schema=new_schema)
```

Comment:
I know we weren't doing this before, but this seems like a good candidate for `validate=False`